### ISSUE TITLE
Cache invalidation strategy for CSS, JS, etc.

### ISSUE THREAD

**Issue by jasonrhodes** [bigboringsystem/bigboringsystem#184]
Possibly related to #180 I think?

Right now it looks like changes to public/js/* and public/css/* might not take effect when we want them because we don't do any versioning on those files. I've seen `file.js?version=<timehash>` or `file.<timehash>.js` strategies, or you could probably update nginx to put super short expires headers on those file types, although that's not awesome. I assume we'd have to manage this manually or else regex the jade files to update references... 

I think I've also seen in Apache you can rewrite requests to some regex like `/(.*)\.[0-9]*\.(js|css)/ ` to omit the timestamp portion like `$1.$2`, but you still have to update the references in the HTML somehow...

Other ideas? Simplest is best, obviously...

**Comment by jasonrhodes** [comment]
Oh hey this is pretty fucking awesome, I wonder if there's a hapi version...

https://www.npmjs.com/package/express-asset-versions

**Comment by isaacs** [comment]
We could just do `?#{time}` and set that to the server start-up time on the static assets.  That'd be enough, and probably doable without writing a whole new Hapi plugin.

**Comment by isaacs** [comment]
This module seems apropos: http://npm.im/hapi-cache-buster/

**Comment by jasonrhodes** [comment]
+1 that sounds good enough to me

**Comment by isaacs** [comment]
I'm not sure that package version is the best indicator, though.  Server start time is a better indicator, since we don't bump the package version for each release.

**Comment by jasonrhodes** [comment]
Yeah, definitely. But whoever does this should be able to borrow the syntax for hijacking the reply and appending a value so that it's available to all views.

**Comment by imakewebthings** [comment]
That plugin allows you to pass anything as the cache buster string, package version is the default if nothing is specified. I'll open a PR using the plugin + @isaacs's server start time suggestion.

**Comment by ChiperSoft** [comment]
Couldn't server start time cause a problem if/when the site starts needing multiple server processes? Different processes reporting different files?

**Comment by imakewebthings** [comment]
@ChiperSoft This is true. I'm still going to open this PR because it's better than nothing. The if you describe is very large and very if right now, and the worst case scenario is somebody downloads the same file twice instead of a cache hit.

If we want to do it "right" we do a hash of the file contents for each asset file (maybe during server startup) and use that checksum as the cache buster. That's a bit more work that I think will not be worth it until (if ever) we get to > 1 app server.
